include/language.php

```diff
@@ -597,10 +597,14 @@ public function get_display_flag() {
 	public function get_display_flag_url() {
 		$flag_url = empty( $this->custom_flag_url ) ? $this->flag_url : $this->custom_flag_url;
 
-		/**
-		 * Let's use `site_url()` so the returned URL will be filtered properly according to the current domain.
-		 */
-		return site_url( set_url_scheme( $flag_url, 'relative' ) );
+		if ( ( defined( 'PLL_CACHE_LANGUAGES' ) && ! PLL_CACHE_LANGUAGES ) || ( defined( 'PLL_CACHE_HOME_URL' ) && ! PLL_CACHE_HOME_URL ) ) {
+			/**
+			 * Let's use `site_url()` so the returned URL will be filtered properly according to the current domain.
+			 */
+			return site_url( set_url_scheme( $flag_url, 'relative' ) );
+		}
+
+		return $flag_url;
```

Comment:
It seems that you misunderstood the purpose of the function. We can not cache the flag url. At least not in the context proposed by this test. The reason is that the url of the flags will change depending on the domain we are visiting.

Ex: 2 languages, `mysite.com` (default), `monsite.fr`:
`$flag_url` will be `mysite.com/wp-content/plugins/polylang/flags/us.png` for English and `mysite.com/wp-content/plugins/polylang/flags/fr.png` for French.
These 2 urls are correct when we display the language switcher on `mysite.com`.
But they must be switched to `monsite.fr/wp-content/plugins/polylang/flags/us.png` and `monsite.fr/wp-content/plugins/polylang/flags/fr.png when we display the language switcher on `monsite.fr`. 

So the urls are dependent on the domain displayed. This is true whether the languages are cached or not. I propose to revert this change.

Reply:
Indeed, that's not exactly the same thing.
Reverted.

Reply:
Oops I didn't thought about it that way ...